Sources/DiffableDataSource/DiffableTableViewDataSource+Replace.swift

```diff
@@ -10,30 +10,42 @@ import UIKit
 extension DiffableTableViewDataSource {
     // MARK: - Public API
 
-    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) throws {
-        try checkUniqueDiffableSections(sections: sections)
+    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) {
+        switch checkUniqueDiffableSections(sections: sections) {
+        case let .success(sections):
+            self.unsafeReplace(sections: sections, animatingDifferences: animatingDifferences)
 
-        var snapshot = NSDiffableDataSourceSnapshot<SectionID, RowID>()
-
-        snapshot.appendSections(sections.map(\.identifier))
-
-        sections.forEach {
-            snapshot.appendItems($0.rows, toSection: $0.identifier)
+        case let .failure(error):
+            logFault(error: error)
         }
-
-        apply(snapshot, animatingDifferences: animatingDifferences)
     }
 
-    public func checkUniqueDiffableSections(sections: [DiffableSection<SectionID, RowID>]) throws {
+    public func checkUniqueDiffableSections(
```

Comment:
[nit] Since the first input parameter is already called `sections` maybe we could rename it like this:

```suggestion
    public func checkUniqueDiffable(
```

Feel free to ignore btw